modules/spring-config/src/main/java/org/mule/config/spring/SpringRegistryLifecycleManager.java

```diff
@@ -45,28 +65,45 @@ protected void registerPhases()
     // Spring custom lifecycle phases
     // ///////////////////////////////////////////////////////////////////////////////////
 
-    /**
-     * A lifecycle phase that will delegate any lifecycle invocations to a container
-     * such as Spring or Guice
-     */
-    class SpringContextInitialisePhase extends ContainerManagedLifecyclePhase
+    class SpringContextInitialisePhase extends MuleContextInitialisePhase
     {
+
         public SpringContextInitialisePhase()
         {
-            super(Initialisable.PHASE_NAME, Initialisable.class, Disposable.PHASE_NAME);
-            registerSupportedPhase(NotInLifecyclePhase.PHASE_NAME);
+            super();
+
+            Set<LifecycleObject> initOrderedObjects = new LinkedHashSet<>();
+            initOrderedObjects.add(new NotificationLifecycleObject(ObjectStoreManager.class));
+            initOrderedObjects.add(new NotificationLifecycleObject(Config.class));
+            initOrderedObjects.add(new NotificationLifecycleObject(Connector.class));
+            initOrderedObjects.add(new NotificationLifecycleObject(Agent.class));
+            initOrderedObjects.add(new NotificationLifecycleObject(Model.class));
+            initOrderedObjects.add(new NotificationLifecycleObject(FlowConstruct.class));
+            initOrderedObjects.add(new NotificationLifecycleObject(Initialisable.class));
+            setOrderedLifecycleObjects(initOrderedObjects);
+
+            setIgnoredObjectTypes(new Class[] {
+                    Component.class,
+                    MessageSource.class,
+                    AbstractMessageProcessorOwner.class,
+                    MessagingExceptionHandler.class,
+                    AbstractAsyncRequestReplyRequester.class,
+                    OutboundRouter.class,
+                    MessageProcessorChain.class,
+                    OutboundRouterCollection.class,
+                    MuleContext.class
+            });
         }
 
-        /**
-         * We don't need to apply any lifecycle here since Spring manages that for us
-         * 
-         * @param o the object apply lifecycle to. This parameter will be ignorred
-         * @throws LifecycleException never thrown
-         */
         @Override
         public void applyLifecycle(Object o) throws LifecycleException
         {
-            // Spring starts initialised, do nothing here
+            if (o instanceof SpringRegistry)
```

Comment:
Should not be this class ignored using setIgnoredObjectTypes instead of using this condition?